packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -75,6 +74,32 @@ export default class CommentsRepo extends AbstractRepo<DbComment> {
     `, [limit]);
   }
 
+  async getPopularComments({minScore = 15, offset = 0, limit = 3}: {
+    offset?: number,
+    limit?: number,
+    minScore?: number,
+  }): Promise<DbComment[]> {
+    return this.any(`
+      SELECT *
+      FROM (
+        SELECT DISTINCT ON ("postId") *
+        FROM "Comments"
+        WHERE
+          CURRENT_TIMESTAMP - "postedAt" < '1 week'::INTERVAL AND
+          "shortform" IS NOT TRUE AND
+          "baseScore" >= $1 AND
+          "retracted" IS NOT TRUE AND
+          "deleted" IS NOT TRUE AND
+          "deletedPublic" IS NOT TRUE AND
+          "needsReview" IS NOT TRUE
+        ORDER BY "postId"
+      ) c
+      ORDER BY fm_comment_confidence(c."_id", 3) DESC
```

Comment:
Does `fm_comment_confidence()` get called for every comment every time this is run?

Reply:
Only the comments that are returned by the sub-query, which is generally ~40. I was planning to add a cache on top of this too in a separate PR before rolling out to all users, but it was extremely simple so I just went ahead and added it here.